lib/presentation/common/widget/horizontal_radio_group_widget.dart

```diff
@@ -0,0 +1,125 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:flutterapp/presentation/common/viewdata/radio_group_item_view_data.dart';
+import 'package:flutterapp/presentation/productDetails/product_details_dimens.dart';
+
+/// A widget that laying it's elements horizontally in a row and divides an available
+/// space evenly between them. Accepts a list with pre-populated [RadioGroupItemViewData].
+/// One item in the group is constructed of:
+/// - border with [ProductDetailsDimens.horizontalRadioButtonBorderWidth] width
+/// and [ProductDetailsDimens.horizontalRadioButtonBorderRadius] radius
+/// - icon (if present) at the left
+/// - one or two lines of text in the middle
+///
+/// Items in the widget may be one of two states: checked or unchecked.
+/// Only one item is allowed to checked at the time.
+///
+/// Widgets up in the chain can receive a checked item 0-based index via [onButtonPressed] callback.
+class HorizontalRadioGroupWidget extends StatefulWidget {
+
+  final Function(int) onButtonPressed;
+  final List<RadioGroupItemViewData> items;
+  final int _defaultCheckedItem = 0;
+
+  HorizontalRadioGroupWidget(this.onButtonPressed, this.items);
+
+  @override
+  State createState() {
+    return _HorizontalRadioGroupWidgetState(
+        _defaultCheckedItem, onButtonPressed, items);
+  }
+}
+
+class _HorizontalRadioGroupWidgetState
+    extends State<HorizontalRadioGroupWidget> {
+
+  int _checkedIndex = -1;
+  final Function(int) onButtonPressed;
+  List<RadioGroupItemViewData> items;
+
+  _HorizontalRadioGroupWidgetState(this._checkedIndex, this.onButtonPressed,
+      this.items);
+
+  void _onChildPressed(int index) {
+    setState(() {
+      _checkedIndex = index;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    List<Widget> children = List();
+    for (int i = 0; i < items.length; i++) {
+      children.add(Flexible(
+        flex: 1,
+        child: GestureDetector(
```

Comment:
Why do we need GestureDetector ? onTap is from it ?

Reply:
yes, to detect onTap